### PR TITLE
DATA: Occupation Model & Migration

### LINKED ISSUE #175: DATA: Occupation Model & Migration
Model - https://github.com/Mums-Who-Code/SCMS.Services/blob/main/SCMS.Services.Api/Designs/Occupation.jpg

### PR DESCRIPTION
**Pull request by muhammadmnorouzi**
closes #175

### CODE REVIEW

File: SCMS.Services.Api/Brokers/Storages/StorageBroker.Occupation.References.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+namespace SCMS.Services.Api.Brokers.Storages
+{
+    public partial class StorageBroker
+    {
+        public void SetOccupationReferences(ModelBuilder modelBuilder)
+        {
+            modelBuilder.Entity<Occupation>()
+                .HasOne(occupation => occupation.Guardian)
+                .WithOne(guardian => guardian.RegisteredOccupation)
```
Comment (ShriHumrudha) [inline SCMS.Services.Api/Brokers/Storages/StorageBroker.Occupation.References.cs:16]: A guardian can have more than one occupation - Teacher, Author, etc.

**Review by ShriHumrudha — CHANGES_REQUESTED** [review]

File: SCMS.Services.Api/Brokers/Storages/StorageBroker.Occupation.References.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+using Microsoft.EntityFrameworkCore;
+using SCMS.Services.Api.Models.Foundations.Occupations;
+
+namespace SCMS.Services.Api.Brokers.Storages
+{
+    public partial class StorageBroker
+    {
+        public void SetOccupationReferences(ModelBuilder modelBuilder)
+        {
```
Comment (ShriHumrudha) [inline SCMS.Services.Api/Brokers/Storages/StorageBroker.Occupation.References.cs:13]: The Guardian and Occupations relation is missing here.

**Review by ShriHumrudha — CHANGES_REQUESTED** [review]

File: SCMS.Services.Api/Brokers/Storages/StorageBroker.Occupation.References.cs (hunk 2 above)
Comment (muhammadmnorouzi) [reply SCMS.Services.Api/Brokers/Storages/StorageBroker.Occupation.References.cs:13]: Will it end with .OnDelete(DeleteBehavior.NoAction); ?

File: SCMS.Services.Api/Migrations/20211201180427_AddOccupationModelRelations.cs [hunk 3]
```diff
@@ -0,0 +1,66 @@
+﻿using System;
```
Comment (ShriHumrudha) [inline SCMS.Services.Api/Migrations/20211201180427_AddOccupationModelRelations.cs:1]: Add copyrights header.

**Review by ShriHumrudha — CHANGES_REQUESTED** [review]

**Review by ShriHumrudha — APPROVED** [review]

**Pull request merged**